src/utils/Cypher.ts

```diff
@@ -161,4 +167,82 @@ export default class Cypher {
     // Save
     await SettingStorage.saveSettings(tenantID, settingsToSave);
   }
+
+  private static async getCryptoSettings(tenantID: string): Promise<CryptoSettings> {
+    const cryptoSettings = await SettingStorage.getCryptoSettings(tenantID);
+    if (!cryptoSettings || !cryptoSettings.crypto) {
+      throw new BackendError({
+        source: Constants.CENTRAL_SERVER,
+        module: MODULE_NAME,
+        method: 'getCryptoSetting',
+        message: `Tenant ID '${tenantID}' does not have crypto settings.`
+      });
+    }
+    return cryptoSettings;
+  }
+
+  private static async migrateSettings(tenantID: string, cryptoSetting: CryptoSetting): Promise<void> {
+    const settingsToMigrate = await Cypher.getSettingsWithSensitiveData(tenantID);
+    // If tenant has settings with sensitive data, migrate them
+    if (!Utils.isEmptyArray(settingsToMigrate)) {
+      // Migrate
+      for (const settingToMigrate of settingsToMigrate) {
+        // Migration already done?
+        if (Utils.isEmptyArray(settingToMigrate.formerSensitiveData)) {
+          // Save former senitive data in setting
+          const formerSensitiveData = Cypher.backupSensitiveData(settingToMigrate);
+          formerSensitiveData.push(Cypher.hash(cryptoSetting.formerKey));
+          settingToMigrate.formerSensitiveData = formerSensitiveData;
+          // Decrypt sensitive data with former key and key properties
+          await Cypher.decryptSensitiveDataInJSON(tenantID, settingToMigrate, true, cryptoSetting);
+          // Encrypt sensitive data with new key and key properties
+          await Cypher.encryptSensitiveDataInJSON(tenantID, settingToMigrate, false, cryptoSetting);
+          // Save setting with sensitive data encrypted with new key
+          await SettingStorage.saveSettings(tenantID, settingToMigrate);
+        }
+      }
+    }
+  }
+
+  private static async getSettingsWithSensitiveData(tenantID: string): Promise<SettingDB[]> {
+    // Get all settings per tenant
+    const settings = await SettingStorage.getSettings(tenantID, {},
+      Constants.DB_PARAMS_MAX_LIMIT);
+    // Filter settings with sensitiveData
+    return settings.result.filter((value: SettingDB) => {
+      if (value?.sensitiveData && !Utils.isEmptyArray(value?.sensitiveData)) {
+        return true;
+      }
+    });
+  }
+
+  private static backupSensitiveData(setting: SettingDB): string[] {
+    const formerSensitiveData: string[] = [];
+    for (const property of setting.sensitiveData) {
+    // Check that the property does exist otherwise skip to the next property
+      if (Utils.objectHasProperty(setting, property)) {
+        const value = _.get(setting, property) as string;
+        // If the value is undefined, null or empty then do nothing and skip to the next property
+        if (value && value.length > 0) {
+          formerSensitiveData.push(value);
```

Comment:
Not only the value but also the key and in a Json doc:
`formerSensitiveData[property] = value;`